### PR TITLE
feat(render): add support for pwsh

### PR DESCRIPTION
**Pull request by phorcys420**
espanso_pwsh

---

This PR adds support for PowerShell v6+.
Since PowerShell v6.0.0, the executable name changed from powershell.exe to pwsh.exe, this PR implements the new `pwsh` interpreter.

Tested working on : 
- Linux (Ubuntu 22.04.3 LTS, XFCE)
- macOS (14.2.1 23C71)
  - Both with and without PowerShell as the default shell (for the PATH magic)

Untested on Windows.

### CODE REVIEW

**Comment by phorcys420** [comment]
whoops, pushed the wrong code.
I have added the macOS-specific code and it builds on Linux but I have yet to test it on macOS, if someone can test, that'd be great !

Here are the test steps :
1. Install [pwsh](https://learn.microsoft.com/en-us/powershell/scripting/install/installing-powershell-on-macos?view=powershell-7.4#install-the-latest-stable-release-of-powershell)
2. Add the following match configuration
```yml
matches:
  - trigger: ":shell"
    replace: "{{output}}"
    vars:
      - name: output
        type: shell
        params:
          shell: pwsh
          cmd: |
            Write-Host "Hello from PowerShell $($PSVersionTable.PSVersion)"
```
3. Start espanso
4. Type ":shell"

**Comment by phorcys420** [comment]
I'll test for Windows later today or tomorrow.

**Comment by phorcys420** [comment]
Is there any code formatting/linting steps ?
I couldn't find anything in the makefile but maybe I'm missing something.

**Comment by AucaCoyan** [comment]
No, as far of Rust, none. Only the test pass
I don't know if there is any requirement in C++ files, but I believe not

File: espanso-render/src/extension/shell.rs [hunk 1]
```diff
@@ -31,6 +31,7 @@ use thiserror::Error;
 pub enum Shell {
   Cmd,
   Powershell,
+  Pwsh,
```
Comment (AucaCoyan) [inline espanso-render/src/extension/shell.rs]: Can I have a comment here indicating powershell v7+ and classic powershell?
or `Windows Powershell` and `Powershell` (core?) as it is named on the docs

**Review by AucaCoyan — APPROVED** [review]
all good!
just left a comment for clarity

File: espanso-render/src/extension/shell.rs (hunk 1 above)
Comment (phorcys420) [reply espanso-render/src/extension/shell.rs]: Sure, done in 362b0bc.

**Comment by phorcys420** [comment]
whoops, I didn't mean to mark it as ready for review.
@AucaCoyan I think we should wait confirmation that it works on macOS and then you can approve the workflow.

**Comment by AucaCoyan** [comment]
Don't worry! You can also click on the cog and my name and the review request should be gone
image
(at least for myself it works)
I'll tag Federico when you are ready to merge 😄 . Take all your time and tests you need!

**Comment by federico-terzi** [comment]
@phorcys420 thanks! Looks great 🚀 

Before proceeding, could you please fix formatting (as in the CI failure) and make sure it works on the target operating systems? :)

**Comment by phorcys420** [comment]
> Could you please fix formatting (as in the CI failure)

I'll look into it, I didn't run any formatting because I didn't see anything about it.
Maybe it'd be nice to add it to a CONTRIBUTING.md file ?

> make sure it works on the target operating systems? :)

yes, that is planned !

**Comment by phorcys420** [comment]
Tested working on macOS 14.2.1 (23C71) both with and without PowerShell as the default shell (for the PATH magic).

https://github.com/espanso/espanso/assets/57866459/681bf9da-d69d-4820-a411-cfa685d42341

**Review by AucaCoyan — APPROVED** [review]
All good!

**Comment by AucaCoyan** [comment]
> I'll look into it, I didn't run any formatting because I didn't see anything about it.
Maybe it'd be nice to add it to a CONTRIBUTING.md file ?

I made the first draft! #1829

**Comment by federico-terzi** [comment]
Thank you 🚀

**Pull request merged**